### PR TITLE
Add linting rules

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
I tried adding pedantic linting rules, however, the analysis server did not like it for some reason, which made it impossible for me to run that analysis locally.

So I decided to add some rules that make sense for this package and correct resulting analysis errors.

 * `directives_ordering` is a fairly general rule, but I think it helps having a good overview over imports.  
  The IntelliJ plugin does this automatically when organizing imports.

 * There are only occurences of `prefer_final_in_for_each` in tests, but there might be for-each loops in the future.

All the other ones apply to most of the code in the package (`cancel_subscriptions` and `close_sinks` only in two cases).

`prefer_const_declarations` is probably the most useful one.

### CODE REVIEW

**Review by axel-op — APPROVED** [review]

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,29 +1,36 @@
 include: package:pedantic/analysis_options.yaml
 linter:
   rules:
-    - unnecessary_const
-    - unnecessary_new
+    unnecessary_const: true
```
Comment (creativecreatorormaybenot) [inline analysis_options.yaml:4]: Needed to exclude the curly braces rule because YAML can only parse lists of one type.

**Comment by creativecreatorormaybenot** [comment]
I will just merge this for now because I want to open another PR. You can adjust linter rules whenever needed, so that should not be a problem.

**Comment by axel-op** [comment]
Another PR???

**Comment by creativecreatorormaybenot** [comment]
@axel-op It's a tiny one. wait a sec

**Pull request merged**